open-vulnerability-clients/src/main/java/io/github/jeremylong/openvulnerability/client/nvd/RateLimitedClient.java

```diff
@@ -121,15 +137,52 @@ class RateLimitedClient implements AutoCloseable {
+        this(maxRetries, minimumDelay, meter, httpClientSupplier, null);
+    }
+
+    /**
+     * Construct a rate limited client with a given delay and request window configuration. This allows callers to
+     * configure 5 requests are allowed over a 30-second rolling window and we will delay at least 4 seconds between
+     * calls to help more evenly distribute the calls across the request window.
+     *
+     * @param maxRetries the maximum number of retry attemps
+     * @param minimumDelay the number of milliseconds to wait between API calls
+     * @param meter the rate meter to limit the request rate
+     * @param httpClientSupplier supplier for custom HTTP clients; if {@code null} a default client will be used
+     * @param cachePath the path to the cache directory, can be null
+     */
+    @SuppressFBWarnings("CT_CONSTRUCTOR_THROW")
+    RateLimitedClient(int maxRetries, long minimumDelay, RateMeter meter, HttpAsyncClientSupplier httpClientSupplier,
+            Path cachePath) {
         this.meter = meter;
         this.delay = minimumDelay;
         LOG.debug("rate limited call delay: {}", delay);
         if (httpClientSupplier == null) {
             NvdApiRetryStrategy retryStrategy = new NvdApiRetryStrategy(maxRetries, minimumDelay);
             SystemDefaultRoutePlanner planner = new SystemDefaultRoutePlanner(ProxySelector.getDefault());
-            client = HttpAsyncClients.custom().setRoutePlanner(planner).setRetryStrategy(retryStrategy)
-                    .useSystemProperties().build();
+            if (cachePath == null) {
+                client = HttpAsyncClients.custom().setRoutePlanner(planner).setRetryStrategy(retryStrategy)
+                        .useSystemProperties().build();
+            } else {
+                File cacheDir = cachePath.toFile();
+                if (!cacheDir.exists() && !cacheDir.mkdirs()) {
+                    throw new NvdApiException("Unable to create http cache directory: " + cacheDir.getAbsolutePath());
+                }
+                LOG.debug("Using http cache directory: {}", cacheDir.getAbsolutePath());
+                CacheConfig cacheConfig = CacheConfig.custom().setMaxCacheEntries(1000).setMaxObjectSize(5_000_000)
+                        .setSharedCache(true).setHeuristicCachingEnabled(true)
+                        .setHeuristicDefaultLifetime(TimeValue.of(Duration.ofHours(20))) // 20 hours caching
+                        .build();
+
+                ResourceFactory resourceFactory = new FileResourceFactory(cacheDir);
+                cacheStorage = new ManagedHttpCacheStorage(cacheConfig);
+
+                client = CachingHttpAsyncClients.custom().setCacheConfig(cacheConfig).setHttpCacheStorage(cacheStorage)
+                        .setResourceFactory(resourceFactory).setRoutePlanner(planner).setRetryStrategy(retryStrategy)
+                        .useSystemProperties().build();
```

Comment:
@jeremylong looking at the codebase of httpclient-cache you should also disable the cache cleanup

```suggestion
                client = CachingHttpAsyncClients.custom().setCacheConfig(cacheConfig).setHttpCacheStorage(cacheStorage)
                        .setDeleteCache(false)
                        .setResourceFactory(resourceFactory).setRoutePlanner(planner).setRetryStrategy(retryStrategy)
                        .useSystemProperties().build();
```

see 
https://github.com/apache/httpcomponents-client/blob/4009567af704530aa0578e34fc63dd8baa945d34/httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/CachingHttpAsyncClientBuilder.java#L137-L139

https://github.com/apache/httpcomponents-client/blob/9941fde19e1737a23264d605ff6eed89552dd8c4/httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/ManagedHttpCacheStorage.java#L200-L214

https://github.com/apache/httpcomponents-client/blob/d7108f60ae8f6d229fcd425a7cfc2010b64a6bf4/httpclient5-cache/src/main/java/org/apache/hc/client5/http/impl/cache/FileResource.java#L105-L110